src/partition/repart.c

```diff
@@ -3640,6 +3640,9 @@ static int partition_populate_directory(Partition *p, const Set *denylist, char
         if (r < 0)
                 return log_error_errno(r, "Failed to create temporary directory: %m");
 
+        if (chmod(root, 0755) < 0)
```

Comment:
one of these days we should rework this all to use open_mkdir_at(), and then operate only via fds...